### ISSUE TITLE
Linux Mint fails to start installation

### ISSUE THREAD

**Issue by banan314** [laurent22/joplin#2906]
## Environment

Joplin version: NA
Platform: Linux (Mint 19.3 Cinnamon)
OS specifics: bash

## Steps to reproduce

1. Run Joplin_install_and_update.sh

**Result:**

Prints
```
     _             _ _       
    | | ___  _ __ | (_)_ __  
 _  | |/ _ \| '_ \| | | '_ \ 
| |_| | (_) | |_) | | | | | |
 \___/ \___/| .__/|_|_|_| |_|
            |_|

Linux Installer and Updater
```
but nothing else happens

## Describe what you expected to happen

Should proceed with installation.

**Comment by rajprakash00** [comment]
Have you tried in this way as written in forum
https://discourse.joplinapp.org/t/cannot-install-in-linux-mint-19-1-mate/1753

**Comment by xmiti** [comment]
Guess the issue is the following command (line 72) ends up with the empty string:
`RELEASE_VERSION=$(wget -qO - "https://api.github.com/repos/laurent22/joplin/releases/latest" | grep -Po '"tag_nam    e": "v\K.*?(?=")')`

**Comment by banan314** [comment]
> Have you tried in this way as written in forum
> https://discourse.joplinapp.org/t/cannot-install-in-linux-mint-19-1-mate/1753

Hi! As I understand the situation on the discourse, the installer did most of its job but failed to create a desktop icon. My case is a complete failure - nothing is done at all :fearful:

**Comment by banan314** [comment]
> Guess the issue is the following command (line 72) ends up with the empty string:
> `RELEASE_VERSION=$(wget -qO - "https://api.github.com/repos/laurent22/joplin/releases/latest" | grep -Po '"tag_nam e": "v\K.*?(?=")')`

Hi @xmiti! What do you suggest? Editing the file is no problem for me, just have to know what to do. First of all, I cannot see an empty string.

**Comment by xmiti** [comment]
Hello @banan314, very interesting.  Have tried that several times and indeed, sometimes the `tag_name` (1.0.195) is displayed, sometimes not:
```text
$ date; wget -qO - "https://api.github.com/repos/laurent22/joplin/releases/latest" | grep -Po '"tag_name": "v\K.*?(?=")'
Fri 27 Mar 2020 12:36:14 PM CET
```
```text
$ date; wget -qO - "https://api.github.com/repos/laurent22/joplin/releases/latest" | grep -Po '"tag_name": "v\K.*?(?=")'
Fri 27 Mar 2020 12:36:25 PM CET
1.0.195
```
Might be an issue related to the JSON formatting.  Using the `jq` could be a dirty fix:
```text
$ date; wget -qO - "https://api.github.com/repos/laurent22/joplin/releases/latest" | jq '.' | grep -Po '"tag_name": "v\K.*?(?=")'
Fri 27 Mar 2020 12:41:41 PM CET
1.0.195
```

**Comment by krlmlr** [comment]
On Ubuntu, I had to download the script to a file and run `bash` on it. I'm seeing a download progress bar, maybe this doesn't work when running the installer script in piped mode?

**Comment by dpoulton-github** [comment]
This has come up on the [forum](https://discourse.joplinapp.org/t/installation-script-output-seems-ok-but-joplin-desktop-not-found/7411/2)

It seems commenting out the Architecture Check allows the script to run.

**Comment by CalebJohn** [comment]
On the [forums](https://discourse.joplinapp.org/t/installation-script-output-seems-ok-but-joplin-desktop-not-found/7411/6?u=calebjohn) one user wase able to get around this by removing the architecture check block. Can anyone confirm this? And if so, what is the output of `uname -a` on your system.

**Comment by CalebJohn** [comment]
For anyone experiencing this issue can you please try the below command?
```
wget -O - https://raw.githubusercontent.com/CalebJohn/joplin/arch-check/Joplin_install_and_update.sh | bash
```

And copy/paste the output

I'm having trouble reliably duplicating the issue

**Comment by mpern** [comment]
The issue was (at least for me on Kubuntu) the regex to determine the current version:

```shell
wget -qO - "https://api.github.com/..." | grep -Po '"tag_name": "v\K.*?(?=")'
                                                               ^
                                                               |
                                                       delete this space
```

**Comment by syrte** [comment]
I also met with this problem. The installation fails silently.
Just for your information, I'm unable to connect to api.github.com in my case. It seems blocked by the firewall of my internet provider. If you are in China, you might have the same issue.

```
$ wget -O - "https://api.github.com/repos/laurent22/joplin/releases/latest"
--2020-03-29 12:21:18--  https://api.github.com/repos/laurent22/joplin/releases/latest
Resolving api.github.com (api.github.com)... 13.250.168.23
Connecting to api.github.com (api.github.com)|13.250.168.23|:443... connected.
Unable to establish SSL connection.
```
Also, I can not access this URL in my browser.

**Comment by LecygneNoir** [comment]
Hello,

I confirm the same problem here on Fedora 31

@CalebJohn for the output of your command:
```
     _             _ _       
    | | ___  _ __ | (_)_ __  
 _  | |/ _ \| '_ \| | | '_ \ 
| |_| | (_) | |_) | | | | | |
 \___/ \___/| .__/|_|_|_| |_|
            |_|

Linux Installer and Updater

Checking architecture...
Checking latest version...
```
No more output, give up the shell silently after the "Checking latest version"

I was unable to have the json version working even relaunching multiple time, but the trick from @mpern works!

Downloading the script without piping and removing the space :
 ```
RELEASE_VERSION=$(wget -qO - "... | grep -Po '"tag_name": "v\K.*?(?=")')
                                                                                                   ^
                                                                                                    |
                                                                                      delete this space
```

After that the script get the version correctly and upgrade Joplin.

I guess github should have changed something in the API :thinking: 

Thanks for the trick! (And for Joplin :hugs: )

**Comment by CalebJohn** [comment]
Thanks everyone for the replies, the fix is on the way!

**Comment by CalebJohn** [comment]
fixed by #2913
